### ISSUE TITLE
bug: tfsec crashes since last saturday

### ISSUE THREAD

**Issue by favoretti** [aquasecurity/tfsec#1446]
Can't figure out on which resource exactly, but one of azure keyvaults.

```
panic: not a string

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.AsString({{{0x0, 0x0}}, {0x0, 0x0}})
	/home/runner/work/tfsec/tfsec/vendor/github.com/zclconf/go-cty/cty/value_ops.go:1256 +0x138
github.com/aquasecurity/tfsec/internal/pkg/adapter/azure/keyvault.adaptSecret(0xc000823220)
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/azure/keyvault/adapt.go:113 +0x165
github.com/aquasecurity/tfsec/internal/pkg/adapter/azure/keyvault.(*adapter).adaptVaults(0xc0025f6130, {0xc000980800, 0xbbdc07, 0x80})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/azure/keyvault/adapt.go:44 +0x347
github.com/aquasecurity/tfsec/internal/pkg/adapter/azure/keyvault.Adapt({0xc000980800, 0x45, 0x80})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/azure/keyvault/adapt.go:18 +0x11c
github.com/aquasecurity/tfsec/internal/pkg/adapter/azure.Adapt({_, _, _})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/azure/adapt.go:30 +0x356
github.com/aquasecurity/tfsec/internal/pkg/adapter.Adapt({0xc000980800, 0x45, 0x80})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/adapt.go:20 +0x7e
github.com/aquasecurity/tfsec/internal/pkg/scanner.(*Scanner).Scan(0xc001ad7080, {0xc000980800, 0x0, 0x80})
	/home/runner/work/tfsec/tfsec/internal/pkg/scanner/scanner.go:51 +0x9f
github.com/aquasecurity/tfsec/internal/app/tfsec/cmd.glob..func1(0x12801a0, {0xc0003e8720, 0x1, 0x6})
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/cmd/root.go:114 +0x7ec
github.com/spf13/cobra.(*Command).execute(0x12801a0, {0xc00001e080, 0x6, 0x6})
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x12801a0)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/home/runner/work/tfsec/tfsec/cmd/tfsec/main.go:11 +0x25
##[error]Bash exited with code '2'.

```

**Comment by liamg** [comment]
Thanks for the report, will look at this in the next hour and provide a fix asap.

**Comment by owenrumney** [comment]
Hey @favoretti  - can you share how you creating the expiration date value? 

I assumed it was using `formatdate` but that works fine in the test using 

```hcl
 resource "azurerm_key_vault_key" "good_example" {
   name         = "generated-certificate"
   key_vault_id = azurerm_key_vault.example.id
   key_type     = "RSA"
   key_size     = 2048
   expiration_date = formatdate("DD MMM YYYY hh:mm ZZZ", "2018-01-02T23:12:01Z")
 
   key_opts = [
     "decrypt",
     "encrypt",
     "sign",
     "unwrapKey",
     "verify",
     "wrapKey",
   ]
 }
```

**Comment by favoretti** [comment]
We do not, that's the whole thing... Here's what it crashes on:
![image](https://user-images.githubusercontent.com/805046/151797578-c5a1284d-130c-4694-81bb-9760f2924446.png)

**Comment by favoretti** [comment]
![image](https://user-images.githubusercontent.com/805046/151797680-df51a13d-dbaa-4ac8-b0e3-405b9e05c639.png)

**Comment by favoretti** [comment]
Just re-tested with fix commit - works, thanks! If you could release it - it will fix all of our pipelines, much appreciated!

**Comment by liamg** [comment]
Awesome, thanks for the confirmation. Released in `v1.0.4` :)
